### PR TITLE
Exploit new support in `firefly-signer` and `firefly-common` for large & scientific JSON numbers

### PR DESCRIPTION
**Pull request by matthew1001**
This PR does 2 things:

1. Pulls in the latest version of:
   - `firefly-common` (see especially PR https://github.com/hyperledger/firefly-common/pull/147)
   - `firefly-signer` (see especially PR https://github.com/hyperledger/firefly-signer/pull/76)
   - `firefly-transaction-manager` (see especially PR https://github.com/hyperledger/firefly-transaction-manager/pull/128)
3. Adds unit tests that exercise the new function introduced by those PRs

### CODE REVIEW

File: internal/ethereum/deploy_contract_prepare.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func (c *ethConnector) prepareDeployData(ctx context.Context, req *ffcapi.Contra
 	ethParams := make([]interface{}, len(req.Params))
 	for i, p := range req.Params {
 		if p != nil {
-			err := json.Unmarshal([]byte(*p), &ethParams[i])
+			err := p.Unmarshal(ctx, &ethParams[i])
```
Comment (EnriqueL8) [inline internal/ethereum/deploy_contract_prepare.go:98]: For context: The type of `p` is `*fftypes.JSONAny` which has implemented an Unmarshal method

File: internal/ethereum/deploy_contract_prepare_test.go [hunk 2]
```diff
@@ -39,19 +40,91 @@ const samplePrepareDeployTX = `{
 	"gas": 1000000,
 	"nonce": "111",
 	"value": "12345678901234567890123456789",
-	"contract": "0xfeedbeef",
+	"contract": "0xdeadbeef",
```
Comment (EnriqueL8) [inline internal/ethereum/deploy_contract_prepare_test.go:43]: 🤣

File: internal/ethereum/deploy_contract_prepare_test.go [hunk 3]
```diff
@@ -39,19 +40,91 @@ const samplePrepareDeployTX = `{
...
+		    {
+			    "internalType":" address",
+			    "name": "y",
+			    "type": "address"
+		    },
+		    {
+			    "internalType":" string",
+			    "name": "z",
+			    "type": "string"
+		    }
```
Comment (EnriqueL8) [inline internal/ethereum/deploy_contract_prepare_test.go:60]: the hole nesting looks weird, probably just GitHub

**Review by EnriqueL8 — APPROVED** [review]
Looks good to me

**Review by EnriqueL8 — CHANGES_REQUESTED** [review]
Wait don't we have to fix the invoke contract as well?

**Comment by EnriqueL8** [comment]
You need the same fix in https://github.com/hyperledger/firefly-evmconnect/blob/6aff1e62adfcf4532cd3ccc501f2a7af4c75ea23/internal/ethereum/prepare_transaction.go#L111

File: internal/ethereum/deploy_contract_prepare_test.go (hunk 2 above)
Comment (matthew1001) [reply internal/ethereum/deploy_contract_prepare_test.go:43]: Yeah a slightly cheeky way to ensure the asserts in the tests don't match on the contract bytes when checking for integers that match `0xdeadbeef`

File: internal/ethereum/deploy_contract_prepare_test.go (hunk 3 above)
Comment (matthew1001) [reply internal/ethereum/deploy_contract_prepare_test.go:60]: yeah I think that's just github

**Comment by matthew1001** [comment]
> You need the same fix in
> 
> https://github.com/hyperledger/firefly-evmconnect/blob/6aff1e62adfcf4532cd3ccc501f2a7af4c75ea23/internal/ethereum/prepare_transaction.go#L111

Good spot, will address that now

**Comment by matthew1001** [comment]
> Could it also affect the gas price
> 
> https://github.com/hyperledger/firefly-evmconnect/blob/6aff1e62adfcf4532cd3ccc501f2a7af4c75ea23/internal/ethereum/send_transaction.go#L99
> 
> ?

I think this case is OK as we're specifically unmarshaling to a `HexInteger` not an `interface{}`

**Comment by matthew1001** [comment]
@EnriqueL8 new commits added to handle the `prepare` path, with unit tests

**Review by EnriqueL8 — APPROVED** [review]
Looks good to me

**Comment by EnriqueL8** [comment]
Re-Running Jobs to upload coverage

**Pull request merged**